obj/empty.go

```diff
@@ -21,22 +17,15 @@ func isEmptyValue(v reflect.Value) bool {
 		return v.Float() == 0
 	case reflect.Interface, reflect.Ptr:
 		return v.IsNil()
+	case reflect.Struct:
+		// Note: we can't rely on a *literal* "is zero" check because
+		// non-zero pointers may still point to *empty* things.
+		for i := 0; i < v.NumField(); i++ {
+			if !isEmptyValue(v.Field(i)) {
```

Comment:
Given the current definition of `isEmptyValue`, this doesn't need a "seen" map for deduplication. We simply don't follow non-nil pointers.